host/linux/aesm.c

```diff
@@ -366,25 +382,31 @@ static oe_result_t _read_response(
         uint32_t size;
 
         /* Get the tag of this payload */
-        if ((pos = _unpack_tag(&envelope, pos, &tag)) == (size_t)-1)
+        if ((pos = (size_t)_unpack_tag(&envelope, pos, &tag)) ==
+            OE_ERROR_UNPACK)
             OE_RAISE(OE_FAILURE);
 
-        if (_make_tag(message_type, WIRE_TYPE_LENGTH_DELIMITED, &tmp_tag) != 0)
+        if (_make_tag(
+                (uint8_t)message_type, WIRE_TYPE_LENGTH_DELIMITED, &tmp_tag) !=
+            0)
             OE_RAISE(OE_FAILURE);
 
         if (tag != tmp_tag)
             OE_RAISE(OE_FAILURE);
 
         /* Get the size of this payload */
-        if ((pos = _unpack_variant_uint32(&envelope, pos, &size)) == (size_t)-1)
+        if ((pos = (size_t)_unpack_variant_uint32(&envelope, pos, &size)) ==
+            OE_ERROR_UNPACK)
             OE_RAISE(OE_FAILURE);
 
         /* Check the size (must equal unread bytes in envelope) */
-        if (size != mem_size(&envelope) - pos)
+        if (size != mem_size(&envelope) - (size_t)pos)
             OE_RAISE(OE_FAILURE);
 
+        size_t temp = (size_t)mem_ptr(&envelope) + pos;
```

Comment:
I think it's probably not good practice to cast pointers into "size_t" to work with pointer arithmetic. Up until recently, I didn't realize that you're not supposed to do pointer arithmetic on (void *) types! But I think doing pointer arithmetic on non-pointer types is awkward.  Instead of casting the return value from the mem_ptr call to (size_t), let's cast it to a (uint8_t *).

So let's change this line to something like:
```
uint8_t * temp = (uint8_t *)mem_ptr(&envelope) + pos;
```